sentry_streams/src/metrics.rs

```diff
@@ -0,0 +1,58 @@
+use crate::kafka_config::PyMetricConfig;
+use sentry_arroyo::metrics::{init, MetricSink, StatsdRecorder};
+use std::net::UdpSocket;
+
+/// A simple UDP sink for sending StatsD metrics
+struct UdpMetricSink {
+    socket: UdpSocket,
+    addr: String,
+}
+
+impl UdpMetricSink {
+    fn new(host: &str, port: u16) -> Result<Self, std::io::Error> {
+        let socket = UdpSocket::bind("0.0.0.0:0")?;
```

Comment:
IIRC this was a bit tricky to get right without any dataloss in prod. i suggest to not use sockets directly and either use statsdproxy (like snuba) or cadence. at least `nonblocking` is necessary: https://github.com/getsentry/statsdproxy/blob/709c9cd22a1f3fa97660895f5432a3c7fee2887e/src/middleware/upstream.rs#L27